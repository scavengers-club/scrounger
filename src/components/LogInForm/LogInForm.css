.login-form {
  display: flex;
  flex-direction: column;
  gap: 2vh;
  width: 80vw;
  margin: 10%;
  align-items: center;
  text-align: center;
}

.login-form input {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  background-color: rgba(255, 255, 255, 0.527);
  color: black;
  border-radius: 7px;
  padding: 3px;
}

.login-form button {
  width: 20vw;
  background-color: var(--purple);
}

.login-form input::placeholder {
  color: var(--purple);
}
