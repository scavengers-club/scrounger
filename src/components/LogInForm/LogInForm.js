import { Link } from 'react-router-dom';

import './LogInForm.css';

export default function LogInForm({ email, setEmail, password, setPassword, handleSubmit }) {
  return (
    <>
      <form className="login-form">
        <label>
          email:
          <input
            type="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            placeholder="enter your email"
          />
        </label>
        <label>
          password:
          <input
            type="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            placeholder="enter your password"
          />
        </label>
        <button onClick={handleSubmit}>Log In</button>
      </form>
      <p>New to Scrounger?</p>
      <Link to="/register">Register</Link>
    </>
  );
}
