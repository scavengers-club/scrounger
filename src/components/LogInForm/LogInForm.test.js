import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import LogInForm from './LogInForm';

test('<LogInForm/> should display an empty form for returning user', () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/login']}>
      <Route path="/login">
        <LogInForm />
      </Route>
    </MemoryRouter>
  );

  screen.getAllByDisplayValue('');
  expect(container).toMatchSnapshot();
});
