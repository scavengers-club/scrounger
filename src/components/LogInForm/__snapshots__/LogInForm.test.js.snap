// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> should welcome current user 1`] = `
<div>
  <form
    class="login-form"
  >
    <label>
      email:
      <input
        placeholder="enter your email"
        type="email"
        value=""
      />
    </label>
    <label>
      password:
      <input
        placeholder="enter your password"
        type="password"
        value=""
      />
    </label>
    <button>
      Log In
    </button>
  </form>
  <div
    class="redirect"
  >
    <p>
      New to Scrounger?
    </p>
    <a
      href="/register"
    >
      Register
    </a>
  </div>
</div>
`;
